timeseries/src/autogluon/timeseries/models/chronos/pipeline/utils.py

```diff
@@ -1,14 +1,210 @@
+            dummy_value=np.nan,
+        )
+
+    def _infinite_iterable(self, data: Iterable[dict]):
+        while True:
+            yield from data
+
+    def _create_training_data(self, data: Iterable[dict]):
+        data = self._infinite_iterable(data)
+        split_transform = self._create_instance_splitter("training")
+        data = split_transform.apply(data, is_train=True)
+        return data
+
+    def _create_validation_data(self, data: Iterable[dict]):
+        data = self._create_instance_splitter("validation").apply(data, is_train=False)
+        return data
+
+    def to_chronos_format(self, entry: dict) -> dict:
+        """Converts an entry from GluonTS data format with past and future targets
+        to the HuggingFace format accepted by the original Chronos models using the ChronosTokenizer.
+
+        Parameters
+        ----------
+        entry : dict
+            time series data entry in GluonTS format with ``past_target`` and ``future_target`` keys
+
+        Returns
+        -------
+        dict
+            time series data entry in HuggingFace format with ``input_ids``, ``attention_mask``, and ``labels``
+        """
+        assert self.tokenizer is not None, "A ChronosTokenizer is required to convert data into the Chronos format"
+        past_target = torch.tensor(entry[f"past_{FieldName.TARGET}"]).unsqueeze(0)
+        input_ids, attention_mask, scale = self.tokenizer.context_input_transform(past_target)
+        future_target = torch.tensor(entry[f"future_{FieldName.TARGET}"]).unsqueeze(0)
+        labels, labels_mask = self.tokenizer.label_input_transform(future_target, scale)
+        labels[labels_mask == 0] = -100
+
+        return {
+            "input_ids": input_ids.squeeze(0),
+            "attention_mask": attention_mask.squeeze(0),
+            "labels": labels.squeeze(0),
+        }
+
+    def to_chronos_bolt_format(self, entry: dict) -> dict:
+        """Converts an entry from GluonTS data format with past and future targets
+        to the format accepted by the ChronosBolt models.
+
+        Parameters
+        ----------
+        entry : dict
+            time series data entry in GluonTS format with ``past_target`` and ``future_target`` keys
+
+        Returns
+        -------
+        dict
+            time series data entry in ChronosBolt format with ``context`` and ``target``
+        """
+        past_target = torch.tensor(entry[f"past_{FieldName.TARGET}"])
+        future_target = torch.tensor(entry[f"future_{FieldName.TARGET}"])
+
+        return {"context": past_target, "target": future_target}
+
+    def __iter__(self) -> Iterator:
+        if self.mode == "training":
+            iterable = self._create_training_data(self.gluonts_dataset)
+        elif self.mode == "validation":
+            iterable = self._create_validation_data(self.gluonts_dataset)
+
+        format_transform_fn = self.to_chronos_format if self.tokenizer is not None else self.to_chronos_bolt_format
```

Comment:
along similar lines to the tokenizer comment above, this line also suggests that it should rather be the pipelines who know how to handle their data formats...